mfr/extensions/zip/render.py

```diff
@@ -0,0 +1,53 @@
+import os
+import zipfile
+
+import markupsafe
+from mako.lookup import TemplateLookup
+
+from mfr.core import extension
+from mfr.core.utils import sizeof_fmt
+
+
+class ZipRenderer(extension.BaseRenderer):
+
+    TEMPLATE = TemplateLookup(
+        directories=[
+            os.path.join(os.path.dirname(__file__), 'templates')
+        ]).get_template('viewer.mako')
+
+    def render(self):
+        zip_file = zipfile.ZipFile(self.file_path, 'r')
+
+        return self.TEMPLATE.render(zipped_filenames=self.format_zip(zip_file))
+
+    def format_zip(self, zip_file):
+
+        if type(zip_file) != zipfile.ZipFile:
+            return 'This is not a valid zip file.'
+
+        filelist = [file for file in zip_file.filelist if not file.filename.startswith('__MACOSX')]
+
+        if not filelist:
+            return 'This zip file is empty.'
+
+        message = '<table class="table table-hover">' \
+                  '<thead>' \
+                    '<th>%-46s</th><th>%19s</th><th>%12s</th>' \
+                  '</thead>'\
+                  % ('File Name', 'Modified    ', 'Size')
+
+        for zinfo in filelist:
+            date = "%d-%02d-%02d %02d:%02d:%02d" % zinfo.date_time[:6]
+            message += "<tr><td>%-46s</td> <td>%s</td> <td>%s<td></tr>" %\
+                       (markupsafe.escape(zinfo.filename), date, sizeof_fmt(int(zinfo.file_size)))
+
+        message += '</table>'
+        return message
```

Comment:
I don't know how I missed this before, but this method (`format_zip`) is essentially doing templating in code.  Why do that when we are already using a templating system (mako)?  You could move the HTML into the template.mako, and have `format_zip` build the structure to be fed to it.